### ISSUE TITLE
Can I use NNI to do NAS on my own dataset?

### ISSUE THREAD

**Issue by enniefu** [microsoft/nni#1960]
I want to train a object detection model, like yolo ssd fast rcnn, than use nni to do NAS on this model. Can I do this base on NNI?

**Comment by QuanluZhang** [comment]
@enniefu , yes, you can. The design goal of NAS on NNI is for users to freely tune their model's architecture and use their own dataset.

Let's say you have your object detection model in pytorch which is executable. It uses your own training data. To tune the neural architecture of your model, you may want to replace an operation in a layer with other candidate operations, in this case you can use `LayerChoice` (refer to [here](https://github.com/microsoft/nni/blob/master/docs/en_US/NAS/NasInterface.md) for details). If you want to add skip connections or change the inputs of a layer, you could use `InputChoice`. After your model is modified using `LayerChoice` and `InputChoice`, your model is ready to be input to NNI NAS trainer to do auto search/tuning.

Let's say you want to use ENAS trainer ([here](https://nni.readthedocs.io/en/latest/NAS/ENAS.html) is document for ENAS trainer), you need to provide `model` (i.e., the modified model), `loss`, `metrics`, `reward_function`, `optimizer`, `num_epochs`, `dataset_train`, `dataset_valid` to `EnasTrainer`. Here `dataset_train` and `dataset_valid` are your own dataset.

ENAS trainer is based on weight sharing, there is another type of search approach which does not share weight. If your search space is small, you may want to try this approach which is more stable than weight sharing approaches. Basically, it runs every candidate architecture as a trial from scratch. You can refer to [here](https://github.com/microsoft/nni/blob/master/docs/en_US/NAS/NasInterface.md#classic-distributed-search) for detailed usage.

**Comment by scarlett2018** [comment]
@enniefu - closing the issue as the question has been answered. please feel free to open new issues if there are more questions or issues.

**Comment by kruthikakr** [comment]
Can i use NNI for NAS for object detection task which is possible with  models like YOLOV5 . If yes.. How can it be achieved ?

**Comment by kruthikakr** [comment]
Any example is available for this ?
